src/_pytest/deprecated.py

```diff
@@ -106,6 +106,15 @@
     " Replace pytest.warns(None) by simply pytest.warns()."
 )
 
+
+HOOK_LEGACY_MARKING = UnformattedWarning(
+    PytestDeprecationWarning,
+    "The hook {type} {fullname} is not marked using pytest.hook{type}.\n"
+    " please use the pytest.hook{type}(...) decorator instead of pytest.mark \n"
+    " to correctly mark hooks as pytest hooks.\n"
```

Comment:
```suggestion
    "The hook{type} {fullname} uses old-style configuration options (marks or attributes).\n"
    " please use the pytest.hook{type}(...) decorator instead to configure hooks.\n"
```

The last part ("to correctly mark hooks as pytest hooks") is misleading as it conveys that only functions decorated with `pytest.hook*` will be considered hooks, which is not true today, we still consider the `pytest_` prefix.

Reply:
Hmm given we have the options at hand, we might go the extra mile and print the actual change to the user:

```
please use the @pytest.hookspec(firstresult=True) instead to configure hooks.\n"
```